src/services/listService.js

```diff
@@ -44,36 +54,38 @@ export const purchaseItem = (item, token) => {
 
   db.collection('lists')
     .doc(token)
+    .collection('items')
+    .doc(normalizedName)
     .update({
-      [normalizedName]: {
-        name: item.name,
-        frequency: item.frequency,
-        lastPurchased: new Date(),
-        oldPurchased: item.lastPurchased,
-        numberOfPurchases: numberOfPurchases,
-        calculatedEstimate: calculatedEstimate,
-      },
+      name: item.name,
+      frequency: item.frequency,
+      lastPurchased: new Date(),
+      oldPurchased: item.lastPurchased,
+      numberOfPurchases: numberOfPurchases,
+      calculatedEstimate: calculatedEstimate,
     });
 };
 
 export const checkForDuplicateItem = async (token, itemName) => {
-  const currentListData = await db.collection('lists').doc(token).get();
-  const itemList = currentListData.data();
-  if (itemList[itemName]) {
-    return true;
-  }
-  return false;
+  const itemSnapshot = await db
+    .collection('lists')
+    .doc(token)
+    .collection('items')
+    .doc(itemName)
+    .get();
+
+  return itemSnapshot.exists;
```

Comment:
I like this replacement from using a conditional to just letting firestore do the work for us and check if the item exists.